packages/patterns-react/src/patterns/CardSection/CardSection.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+/**
+ * Copyright IBM Corp. 2016, 2018
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useLayoutEffect } from 'react';
+import PropTypes from 'prop-types';
+import root from 'window-or-global';
+import { settings as ddsSettings } from '@carbon/ibmdotcom-utilities';
+import { settings } from 'carbon-components';
+import classNames from 'classnames';
+import { featureFlag, matchHeight } from '@carbon/ibmdotcom-utilities';
+import { DDS_CARD_SECTION } from '../../internal/FeatureFlags';
+import CardsWithImagesGroup from './CardsWithImagesGroup';
+import CardsWithoutImagesGroup from './CardsWithoutImagesGroup';
+
+const { stablePrefix } = ddsSettings;
+const { prefix } = settings;
+
+/**
+ * Cards with images pattern
+ *
+ * @param {object} props props object
+ * @param {Array} props.cardsGroup cardsGroup array with title, groupCard and cards properties
+ * @returns {object} JSX Object
+ */
+const CardSection = ({ theme, cardsGroup, cardType }) => {
+  useLayoutEffect(() => {
+    root.addEventListener(
+      'resize',
+      matchHeight(`.${prefix}--cards-with-images-group__cards`)
+    );
+  });
+
+  /**
+   * sets the class name based on theme type
+   *
+   * @param {string} theme theme type ( g10 | white/default )
+   * @returns {string} theme css class names
+   */
+  const setTheme = theme => {
+    return theme && `${prefix}--cards-with-images--${theme}`;
+  };
+
+  return featureFlag(
+    DDS_CARD_SECTION,
+    <section
+      data-autoid={`${stablePrefix}--cards-with-images`}
+      className={classNames(`${prefix}--cards-with-images`, setTheme(theme))}>
+      <div className={`${prefix}--cards-with-images__container`}>
+        <div className={`${prefix}--cards-with-images__row`}>
+          <div className={`${prefix}--cards-with-images__col`}>
+            {_renderCardsGroup(theme, cardsGroup, cardType)}
+          </div>
+        </div>
+      </div>
+    </section>
+  );
+};
+
+/**
+ * Render Cards with images Group Component
+ *
+ * @private
+ * @param {string} theme theme name
+ * @param {Array} cardsGroup cardsGroup array with title, groupCard and cards properties
+ * @returns {object} JSX Object
+ */
+const _renderCardsGroup = (theme, cardsGroup) => {
+  return cardsGroup.map(group => {
+    let Component;
+    if (group.type === 'simpleCards') {
+      Component = CardsWithoutImagesGroup;
+    } else if (group.type === 'imageCards') {
+      Component = CardsWithImagesGroup;
+    }
```

Comment:
I think to @jeffchew's point we've built this slightly backward.

###### CardSection

```jsx
function CardSection (cards) { 
  <h2>title</h2>
  cards.forEach((card) => {
    return card;
  })
}

export CardSection;
```
I did this quickly to hopefully demonstrate what we're striving to do, so ignore the imperfect syntax. `CardSection` is basically the pattern. You can either pass in an image, or don't pass in an image to the cards it's optional.

For the time being to avoid breaking anyone's code while we figure out naming and more, we can create wrappers that dictate the specific patterns originally created.

###### ImageCards
```jsx
import CardSection;

function ImageCards (data) {
  if images included in data
  <CardSection cards={images included} />
}

export ImageCards;
```

###### SimpleCards
```jsx
import CardSection;

function SimpleCards (data) {
  if no images in data
  <CardSection cards={no images} />
}

export SimpleCards;
```

`ImageCards`, and `SimpleCards` have the dependency on `CardSection`. Not the other way around.


Let me know if we need to jump on a call real quick to talk through this.

Reply:
@photodow do we anticipate `ImageCards` and `SimpleCards` to be two separate patterns that just use the underlying `CardSection` component, or we are introducing a `CardSection` pattern (the intent), with two variations to choose from?

Reply:
I think it kind of depends on how this pattern naming exercise pans out. No matter how you slice the dice the `CardSection` in my opinion has two variants. You can either include the images, or you don't.  These wrappers at the moment buys us some time with Ke and the fragility of Drupal at the moment. If the pattern naming work going on dictates a pattern for `ImageCards`, and `SimpleCards` then the outstanding question from me is will our adopters expect a pattern to be exported with that exact name. If not, then we deprecate these, update Ke and team about the changes. If they are required then using this wrapper method we're still sharing the same code, maybe add a little specific validation on the data, but we're not duplicating code this way at least we've mostly just added a pointer to the core `CardSection` with maybe an opinionated data model. Does that make sense?

Reply:
Got it, and agreed that regardless we need to build out `CardSection` no matter how it's sliced. I like the idea of passing `type` as a prop still, then the cards are rendered out accordingly. This also can open up the ability to add more card types in the future:

```
<CardSection type="image" data={{...}} />
```

If at the end of the day we need to create "decorator" pattern wrappers like `ImageCards` or `SimpleCards`, those can be a lightweight wrapper pattern that uses `CardSection` and explicitly setting the type, and also shaping the PropType shape accordingly based on the pattern type.

Reply:
Okay, I will rewrite the components.

Reply:
> ```jsx
> <CardSection type="image" data={{...}} />

@jeffchew if we want to add the extra level of logic within the CardSection then we could totally go that route. I guess I question if it's necessary. What exactly would it do? Just validate the data and require images or not? The style, layout or function doesn't need to really change I don't think. The image and eyebrow would still need to be passed in via the data. CardLink should handle the optional eyebrow and image. I see it making the wrappers more simple assuming they are necessary in the first place. If it's not necessary and needs to be removed it adds a second touchpoint to remove.

Reply:
To your original point, I guess it all depends on how the naming things shakes out, but going with building out `CardSection` should get us through this for now and we can come back to it.